initialization/initialization.go

```diff
@@ -362,19 +352,17 @@ func (init *Initializer) initFile(computeProviderID uint, fileIndex int, numLabe
 	}
 
 	init.logger.Info("initialization: file #%v completed; number of labels written: %v", fileIndex, numLabelsWritten)
-
 	return nil
 }
 
 func (init *Initializer) updateSessionNumLabelsWritten(numLabelsWritten uint64) {
 	atomic.StoreUint64(&init.numLabelsWritten, numLabelsWritten)
 
-	init.mtx.RLock()
-	ch := init.numLabelsWrittenChan
-	init.mtx.RUnlock()
-
-	if ch != nil {
-		ch <- numLabelsWritten
+	select {
+	case init.numLabelsWrittenChan <- numLabelsWritten:
```

Comment:
This write will `panic` when `init.numLabelsWrittenChan` is closed. It can happen because closing is asynchronous. IMO, sending should be wrapped with a `init.mtx.Lock()`.

Reply:
I'll double check if this causes any possible deadlocks, if not I'll add the protection via mutex

Reply:
A call to `Stop()` triggers the closing of this channel. This happens after all go routines in `initFile` have returned, and thereby all possible callers of `updateSessionNumLabelsWritten` have ended.